packages/datasets-fs/src/commands/generateV2.ts

```diff
@@ -0,0 +1,204 @@
+import { CaptchaTypes, CaptchaWithoutId, Captchas, CaptchasContainerSchema, Item, RawSolution } from '@prosopo/types'
+import { Generate, ArgsSchema as GenerateArgsSchema } from './generate.js'
+import { ProsopoEnvError } from '@prosopo/common'
+import { at, get, lodash } from '@prosopo/util'
+import { blake2AsHex } from '@polkadot/util-crypto'
+import { z } from 'zod'
+import bcrypt from 'bcrypt'
+import cliProgress from 'cli-progress'
+import fs from 'fs'
+
+export const ArgsSchema = GenerateArgsSchema.extend({
+    minCorrect: z.number().optional(),
+    minIncorrect: z.number().optional(),
+    minLabelled: z.number().optional(),
+    maxLabelled: z.number().optional(),
+    count: z.number().optional(),
+})
+export type ArgsSchemaType = typeof ArgsSchema
+export type Args = z.infer<ArgsSchemaType>
+
+export class GenerateV2 extends Generate<ArgsSchemaType> {
+    public override getArgSchema() {
+        return ArgsSchema
+    }
+
+    public override getDescription(): string {
+        return 'Generate distinct captchas producing captcha challenges comprising one or more rounds, mixing labelled and unlabelled data into a single round'
+    }
+
+    public override getOptions() {
+        return lodash().merge(super.getOptions(), {
+            count: {
+                number: true,
+                description: 'Number of captchas to generate',
+            },
+            minCorrect: {
+                number: true,
+                description: 'Minimum number of correct images in each captcha',
+            },
+            minIncorrect: {
+                number: true,
+                description: 'Minimum number of incorrect images in each captcha',
+            },
+            minLabelled: {
+                number: true,
+                description: 'Minimum number of labelled images in each captcha',
+            },
+            maxLabelled: {
+                number: true,
+                description: 'Maximum number of labelled images in each captcha',
+            },
+        })
+    }
+
+    public override async _run(args: Args) {
```

Comment:
_run is a long method here, will be hard to maintain. Might be worth breaking up a little